__device-tests__/helpers/utils.js

```diff
@@ -118,10 +126,92 @@ const stopDriver = async ( driver: wd.PromiseChainWebdriver ) => {
 	}
 };
 
+// attempts to type a string to a given element, need for this stems from
+// https://github.com/appium/appium/issues/12285#issuecomment-471872239
+// https://github.com/facebook/WebDriverAgent/issues/1084
+const typeString = async ( driver: wd.PromiseChainWebdriver, element: wd.PromiseChainWebdriver.Element, str: string, clear: boolean = false ) => {
+	if ( clear ) {
+		await element.clear();
+	}
+
+	if ( isAndroid() ) {
```

Comment:
The need for this additional work for Android is because of how the type API works. It basically clears the field first before start typing. Although Appium implemented a workaround in their additional implementation on top of this to allow appending by copying the existing text before typing the new one, unfortunately, it still performs the 'clear' operation under the hood. Because of that when it clears it, it doesn't know when to stop properly and removes the block, as if the user had touched the backspace button on an empty block. 

The workaround uses `adb shell input <text>` instead to send the text.